config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceUtils.java

```diff
@@ -0,0 +1,62 @@
+package org.hypertrace.config.service;
+
+import com.google.common.base.Strings;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This class contains utility methods.
+ */
+public class ConfigServiceUtils {
+
+  public static final String DEFAULT_CONTEXT = "DEFAULT-CONTEXT";
+
+  private ConfigServiceUtils() {
+    // to prevent instantiation
+  }
+
+  /**
+   * Deep merge the specified {@link Value} configs with overridingConfig taking precedence over
+   * defaultConfig for the same keys.
+   * @param defaultConfig
+   * @param overridingConfig
+   * @return the resulting config obtained after merging defaultConfig and overridingConfig
+   */
+  public static Value merge(Value defaultConfig, Value overridingConfig) {
```

Comment:
Please create unit tests for this merging code specifically, as this is a decent amount of logic.

Reply:
Done